Orange/preprocess/preprocess.py

```diff
@@ -568,3 +568,52 @@ def __call__(self, data):
         for pp in self.preprocessors:
             data = pp(data)
         return data
+
+
+class SmartNormalize(Preprocess):
```

Comment:
We need a better name for this kind of normalization that handles both dense and sparse cases.